app/views/microposts/_micropost.html.erb

```diff
@@ -12,8 +12,8 @@
 											 data: { confirm: "Вы уверены?" } %>
 		<% end %>
 		<!-- I know it wery ugly and this code should be in model or in helper (I guess) -->
-		<% arr = micropost.votes.collect {|i| i[:grades].to_f } %>
-		<p> <%= arr == [] ? 'No votes' : arr.inject{ |sum, el| sum + el }.to_f / arr.size %> </p>
+		<% rating = RatingService.call(micropost.votes) %>
+		<p> <%= rating.failure? ? 'No votes' : rating.value! %> </p>
```

Comment:
Nice catch with not moving that part to service object! 
That's presentation logic, so it has to be in view layer or presenters layer.